apps/core/tests/test_models.py

```diff
@@ -194,14 +209,35 @@ def test_create_omics_unit_with_non_default_status(self):
         self.assertEqual(omics_unit.status, models.OmicsUnit.STATUS_INVALID)
 
     def test_model_representation(self):
-        # TODO
-        # raise NotImplementedError('You have work to do @thomasdenecker!')
-        pass
+
+        omics_unit = models.OmicsUnit.objects.create(
+            reference=self.reference,
+            strain=self.strain,
+            type=self.type,
+        )
+
+        self.assertEqual(str(omics_unit), str(omics_unit.reference))
 
     def test_cannot_create_two_omics_units_with_same_reference_type_and_strain_(self):  # noqa
-        # TODO
-        # raise NotImplementedError('You have work to do @thomasdenecker!')
-        pass
+
+        qs = models.OmicsUnit.objects.all()
+        self.assertEqual(qs.count(), 0)
+
+        models.OmicsUnit.objects.create(
+            reference=self.reference,
+            strain=self.strain,
+            type=self.type
+        )
+
+        with self.assertRaises(IntegrityError):
+            with transaction.atomic():
```

Comment:
Can you explain why those two `with` statements are required? Did you try to remove the second one?

Reply:
The first `with` is used to record exceptions.
The second `with`is used to realise all tests in one time. All tests must be performed otherwise an error will be returned

Reply:
- The first one ensure that the wrapped block will raise an `IntegrityError`
- The second `with` statement ensures [atomic transactions](https://docs.djangoproject.com/en/1.11/topics/db/transactions/#django.db.transaction.atomic) in the wrapped block. Are your familiar with this notion?